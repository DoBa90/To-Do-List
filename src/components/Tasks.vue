<template>
    <div class="tasks">
        <li class="tasks__items" v-for="item in text" :key="item">
            <input class="tasks__input" type="checkbox" :value="item" v-model="checkedProducts"/>
            <Tile 
              class="tasks__tile" 
              :date="item.date" 
              :name="item.name" 
              :image="item.image" 
              :description="item.description" 
            />
        </li>
        <div class="tasks__wrapper" >
          <Button class="tasks__round-button" text="" variant="roundButton" @click="openOverlay()" />
          <Button class="tasks__button" text="LÃ¶schen" variant="whiteButton" @click="deleteTasks()"/>
          <Button class="tasks__button" text="Erledigt" variant="blueButton" />
        </div>
    </div>
</template>

<script>
import Tile from './Tile.vue'
import Button from './Button.vue'

export default {
  name: 'HelloWorld',
  components: { Tile, Button },
  props: {
    text: Array
  },
  data(){
    return {
      checkedProducts:[]
    }
  },
  methods: {
    deleteTasks() {
      console.log("deletedtasks");
      // console.log("deletedtasks", this.checkedProducts)
      this.$emit("deletedtasks", this.checkedProducts);
    },
    openOverlay() {
      this.$emit('activateOverlay', true )
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style lang="scss" scoped>
.tasks {
    &__items {
        color: #000000;
        opacity: 1;
        display: flex;
        flex-direction: row;
        align-items: center;
        gap: 8px;
        padding-bottom: 20px;
    }

    &__wrapper {
        display: grid;
        grid-template-columns: auto auto auto;
        align-items: center;
        column-gap: 10px;
        justify-content: space-around; 
    }

    &__button {
        justify-self: end;
    }
}

@media only screen and (min-width: 768px) {
.tasks {
    &__wrapper {
        grid-template-columns: auto 140px 140px;
        padding-left: 28px;
        justify-content: initial; 
    }
}  
}
</style>
