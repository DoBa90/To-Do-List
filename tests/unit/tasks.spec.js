import { shallowMount } from '@vue/test-utils'
import Tasks from '@/components/Tasks.vue'


const mountComp = options => shallowMount(Tasks, {
    propsData: {
          text: [],
    },
    mocks: options?.mocks
});

describe('Tasks.vue', () => {
    it('renders', () => {
        const tasks = mountComp();

        expect(tasks.exists()).toBeTruthy();
    })

    it('renders variant "default" correctly', () => {
        const tasks = mountComp();

        expect(tasks.html()).toMatchSnapshot();
    });

      // it('renders props.msg when passed', () => {
      //     const tasks = mountComp();
      //     const text = 'Hier alle Geräte ansehen und das richtige Produkt finden';
    
      //     expect(tasks.text()).toMatch(text)
      // })
})



// import { createLocalVue, shallowMount } from '@vue/test-utils';
// import NavigationBox from '@/components/NavigationBox/NavigationBox.vue';

// const localVue = createLocalVue();

// const mountComp = options => shallowMount(NavigationBox, {
//     localVue,
//     propsData: {
//         headline: 'Hier alle Geräte ansehen und das richtige Produkt finden',
//         url: 'https://kaufland.de/elektronik',
//         href: 'https://kaufland.de/elektronik',
//         ...options?.propsData
//     },
//     mocks: options?.mocks
// });

// describe('NavigationBox', () => {
//     it('renders', () => {
//         const navigationBox = mountComp();

//         expect(navigationBox.exists()).toBeTruthy();
//     })