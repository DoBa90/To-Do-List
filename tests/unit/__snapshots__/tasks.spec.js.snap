// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks.vue renders variant "default" correctly 1`] = `
<div class="tasks">
  <div class="tasks__wrapper">
    <button-stub text="" variant="roundButton" class="tasks__round-button"></button-stub>
    <button-stub text="LÃ¶schen" variant="whiteButton" class="tasks__button"></button-stub>
    <button-stub text="Erledigt" variant="blueButton" class="tasks__button"></button-stub>
  </div>
</div>
`;
